cmd/frontend/main.go

```diff
@@ -160,6 +163,17 @@ func forward(logger log.Logger, target *url.URL, transport http.RoundTripper) ht
 	client := http.Client{Transport: transport}
 
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		username := os.Getenv(authUsernameEnv)
+		password := os.Getenv(authPasswordEnv)
+		if len(username) > 0 && len(password) > 0 {
+			level.Info(logger).Log("msg", "AUTH_USERNAME and AUTH_PASSWORD are set, handling request with basic auth")
+			reqUser, reqPass, ok := req.BasicAuth()
+			if !ok || (reqUser != username && reqPass != password) {
```

Comment:
`if !ok || (reqUser != username && reqPass != password)`  
can be changed to 
`if !ok || (reqUser != username || reqPass != password)`

Reply:
good catch, thanks, updated